tock/hours/views.py

```diff
@@ -935,35 +936,50 @@ class ReportingPeriodDetailView(PermissionMixin, ListView):
     template_name = 'hours/reporting_period_detail.html'
     context_object_name = 'timecard_list'
 
+    def dispatch(self, *args, **kwargs):
```

Comment:
Love this! didn't know about the [`Method Flowchart` for `django.views.generic.list.ListView`](https://docs.djangoproject.com/en/1.11/ref/class-based-views/generic-display/#listview).

Reply:
You probably already know this, but https://ccbv.co.uk/ is also excellent for tracing through the CBV steps